data-model-translator-wrapper/src/main/java/ai/aitia/arrowhead/dmtpw/DataModelTranslatorWrapperSystemInfo.java

```diff
@@ -0,0 +1,197 @@
+package ai.aitia.arrowhead.dmtpw;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Component;
+
+import ai.aitia.arrowhead.Constants;
+import eu.arrowhead.common.SystemInfo;
+import eu.arrowhead.common.exception.InvalidParameterException;
+import eu.arrowhead.common.http.filter.authentication.AuthenticationPolicy;
+import eu.arrowhead.common.http.model.HttpInterfaceModel;
+import eu.arrowhead.common.http.model.HttpOperationModel;
+import eu.arrowhead.common.model.InterfaceModel;
+import eu.arrowhead.common.model.ServiceModel;
+import eu.arrowhead.common.model.SystemModel;
+import eu.arrowhead.common.service.validation.name.DataModelIdentifierNormalizer;
+import eu.arrowhead.common.service.validation.name.DataModelIdentifierValidator;
+import eu.arrowhead.dto.enums.ServiceInterfacePolicy;
+
+@Component
+public class DataModelTranslatorWrapperSystemInfo extends SystemInfo {
+
+	//=================================================================================================
+	// members
+
+	private SystemModel systemModel;
+	
+	@Value(DataModelTranslatorWrapperConstants.$TRANSLATION_SCRIPT_LOCATION)
```

Comment:
you should override the customInit() to check whether init and translation scripts are defined and exists, modelIds and mime types lists are not null and not empty.